lib/spack/spack/build_systems/cmake.py

```diff
@@ -67,8 +104,15 @@ class CMakePackage(spack.package_base.PackageBase):
             when="^cmake@3.9:",
             description="CMake interprocedural optimization",
         )
+
+        if sys.platform == "win32":
+            generator("ninja")
+        else:
+            generator("ninja", "make", default="make")
+
         depends_on("cmake", type="build")
-        depends_on("ninja", type="build", when="platform=windows")
+        depends_on("gmake", type="build", when="generator=make")
```

Comment:
I approve of this!

```
$ spack spec gmake +guile | grep cmake
```

has nothing, so it shouldn't break "advanced" gmake builds. `gmake ~guile` doesn't have deps.